engines/native/sandbox.go

```diff
@@ -76,6 +86,39 @@ func newSandbox(b *sandboxBuilder) (*sandbox, error) {
 	return s, nil
 }
 
+func fetchContext(context, destdir string) error {
+	filename, err := util.Download(context, destdir)
+	if err != nil {
+		return fmt.Errorf("Error downloading '%s': %v", context, err)
+	}
+
+	// TODO: verify if this will harm Windows
+	if err = os.Chmod(filename, 0700); err != nil {
+		return fmt.Errorf("Error setting file '%s' permissions: %v", filename, err)
+	}
+
+	unpackedFile := ""
+	switch filepath.Ext(filename) {
+	case ".zip":
+		err = runtime.Unzip(filename)
+	case ".gz":
+		unpackedFile, err = runtime.Gunzip(filename)
+	}
+
+	if err != nil {
+		return fmt.Errorf("Error unpacking '%s': %v", filename, err)
```

Comment:
Don't expose absolute filepath to users... They have no need to know that. The URL is more important.